apps/minifront/src/state/helpers.ts

```diff
@@ -26,49 +79,32 @@ export const plan = async (
   return plan;
 };
 
-export const authWitnessBuild = async (
-  req: PartialMessage<AuthorizeAndBuildRequest>,
-  onStatusUpdate?: (
-    status?: (AuthorizeAndBuildResponse | WitnessAndBuildResponse)['status'],
-  ) => void,
-) => {
-  for await (const { status } of viewClient.authorizeAndBuild(req)) {
-    if (onStatusUpdate) onStatusUpdate(status);
-    switch (status.case) {
-      case undefined:
-      case 'buildProgress':
-        break;
-      case 'complete':
-        return status.value.transaction!;
-      default:
-        console.warn('unknown authorizeAndBuild status', status);
-    }
-  }
-  throw new Error('did not build transaction');
-};
-
-export const witnessBuild = async (
-  req: PartialMessage<WitnessAndBuildRequest>,
+const build = async (
+  req: PartialMessage<AuthorizeAndBuildRequest> | PartialMessage<WitnessAndBuildRequest>,
+  skipAuth: boolean,
   onStatusUpdate: (
     status?: (AuthorizeAndBuildResponse | WitnessAndBuildResponse)['status'],
   ) => void,
 ) => {
-  for await (const { status } of viewClient.witnessAndBuild(req)) {
+  const buildFn = skipAuth ? 'witnessAndBuild' : 'authorizeAndBuild';
```

Comment:
once again, we could use the classification rather than a boolean. also, would prefer to select the actual function instead of an index function.

maybe buildFn should be a parameter, since we have access to the classification in the caller. or the caller could actually use the function and pass the promised response as a parameter

Reply:
JK, I can just use the function's `.name` property. Will do that.